[docs] Add system example for prop + theme key

peteruithoven opened the pull request:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

This PR adds an extra example to the custom style props documentation. 
It took me quite a while to figure out that by using this `themeKey` we also get the logic that in the case of `padding` enables the use of arrays etc. I hope this example will clarify that for others.

eps1lon commented:
Example added in https://deploy-preview-16099--material-ui.netlify.com/system/basics/#style-options-style-function

I guess this is technically explained in the example above but this one is a simpler example. 

Maybe start with this simple one and move the above full example below?

peteruithoven commented:
Good point. I've moved the borderColor example after the gridGap example. I've also added a short description and a usage example to that borderColor example.
